feat(ONB-201): implement keyword generator with n-gram extraction

nagmex210 opened the pull request:
- Add text input interface with cleaning functionality
    - Implement 1-gram, 2-gram, and 3-gram keyword generation
    - Use Set approach for efficient duplicate removal
    - Display keywords as separate tags using Tailwind class names for easy migration
    - Follow Vue 2.2.6 options API and ma- naming conventions

nagmex210 committed:
feat: implement keyword generator with n-gram extraction

    - Add text input interface with cleaning functionality
    - Implement 1-gram, 2-gram, and 3-gram keyword generation
    - Use Set approach for efficient duplicate removal
    - Display keywords as separate tags using Tailwind class names for easy migration
    - Follow Vue 2.2.6 options API and ma- naming conventions

turanibrahim reviewed (changes requested):
Please do changes listed on the comment

nagmex210 committed:
refactor: modernize project architecture with TypeScript and Action Kit

  - Add TypeScript configuration and ESLint with Airbnb rules
  - Restructure components following atomic design principles
  - Convert all components to Composition API with TypeScript
  - Replace custom UI with Action Kit components (MaCard, MaButton, MaBadge)
  - Implement OOCSS methodology using Tailwind @apply directive
  - Add v- prefix to custom components and rename VHeader → VSpecialHeader
  - Set up Vitest testing infrastructure with unit tests
  - Integrate Action Kit global styles and remove redundant code

turanibrahim commented on .eslintrc.cjs:
This config is not correct do it for typescript

turanibrahim commented:
- Please do atomic commits, I will review git history. Please use https://www.conventionalcommits.org/en/v1.0.0/
- All the files and components should be ts
- If you need explanation you can reach out to me directly I can explain

nagmex210 committed:
refactor: address code review feedback and improve project structure

This commit addresses multiple code review feedback items to improve the
overall project architecture and code quality:
  - Renamed VSpecialHeader to VHeader for better naming convention
  - Moved VHomepage and VKeywordGenerator from organisms to pages folder
  - Implemented dynamic layout system with layout constants
  - Updated App.vue to use router-view instead of conditional rendering
  - Enhanced footer with proper styling and sticky bottom positioning
  - Updated ESLint configuration for proper TypeScript support(I hope :))
  - Removed the addressed tests

nagmex210 committed:
Fix: removed scoped css from Vheader

nagmex210 committed:
Fix: removed scoped css from multiple files i tought it was only in Vheader

nagmex210 committed:
refactor: simplify navigation logic and improve code clarity

turanibrahim commented on src/components/organisms/VFooter.vue:
```diff
@@ -0,0 +1,33 @@
+  margin: 0;
+  font-size: 0.875rem;
+  font-weight: 600;
+}
```
Do them with tailwind @apply function use scss. You might need to install post-css to use scss

nagmex210 replied:
abi apply yarısında kullanmışım yarısında kullanmamışım dikkatsizliğme gelmiş Vhomapage de var mesela düzeltirim buraları

turanibrahim commented on src/components/organisms/VHeader.vue:
```diff
@@ -0,0 +1,95 @@
+// Keep tab highlighting in sync with current route
+// This ensures correct tab is highlighted regardless of how user navigates 
+// (clicking tabs, direct URL, browser back/forward buttons)
+watch(
+  () => route.name,
+  (name) => {
+    const tabKey = tabMap[name as keyof typeof tabMap];
+    if (tabKey) {
+      activeTab.value = tabKey;
+    }
+  },
+  { immediate: true }
+);
```
```suggestion
const activeTab = computed(() => {
	const routeName = route.name;
	return tabMap[routeName as keyof typeof tabMap]
});
```

nagmex210 replied:
🔥

turanibrahim commented on src/components/organisms/VHeader.vue:
```diff
@@ -0,0 +1,95 @@
+const router = useRouter();
+const route = useRoute();
+
+const activeTab = ref("home");
```
```suggestion
```

turanibrahim commented on src/components/organisms/VHeader.vue:
```diff
@@ -0,0 +1,95 @@
+
+// Keep tab highlighting in sync with current route
+// This ensures correct tab is highlighted regardless of how user navigates 
+// (clicking tabs, direct URL, browser back/forward buttons)
```
```suggestion
```

nagmex210 replied:
abi bunu kendim hatırlayam diye yazdıydım silem yanlış ise

turanibrahim commented on src/layouts/default.vue:
```diff
@@ -0,0 +1,26 @@
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+}
```
Please use tailwind apply on here

turanibrahim commented on src/layouts/default.vue:
```diff
@@ -0,0 +1,26 @@
+}
+
+.main-content {
+  flex: 1;
```
Please use tailwind apply on here

turanibrahim commented on src/layouts/layout-constants.ts:
Can you move this to under `@/src/constants/layout-constats.ts`

turanibrahim commented on src/pages/VHomepage.vue:
```diff
@@ -0,0 +1,50 @@
+            </p>
+            <div class="button-section">
+              <a href="https://mobileaction.atlassian.net/l/cp/7fw8MroM" target="_blank">
+                <MaButton 
```
Please use kebab case on component usages its a better practice for vue
```suggestion
                <ma-button 
```

turanibrahim commented on src/pages/VHomepage.vue:
```diff
@@ -0,0 +1,50 @@
+              <a href="https://mobileaction.atlassian.net/l/cp/7fw8MroM" target="_blank">
+                <MaButton 
+                    type="primary" 
+                    @click="openTaskLink"
```
```suggestion
```
🚔

nagmex210 replied:
🙉🙉 abi görmemişim vallaha kb

turanibrahim commented on src/pages/VKeywordGenerator.vue:
```diff
@@ -0,0 +1,112 @@
+import { MaButton, MaCard, MaInput, MaBadge } from '@mobileaction/action-kit';
+
+const inputText = ref('');
+const keywords = ref([]);
```
```suggestion
const oneGramKeywords = ref([]);
const twoGramKeywords = ref([]);
const threeGramKeywords = ref([]);
```
My suggestion is use one object reference in `ref` as suggested in vue documentation. When you have complex structures in the ref value you could have reference/reactivity issues while you mutating them.

turanibrahim commented on src/pages/VKeywordGenerator.vue:
```diff
@@ -0,0 +1,112 @@
+  result.push({ type: '3-gram', keywords: [...threeGramSet] });
+  
+  keywords.value = result;
+};
```
Functions always do one think, if they need to multiple things we should compose them together. Long functions consume a lot of cognitive energy so we should split them into small functions. It doesn't make sense this kinda small projects but when you image you need to read hundreds of functions during the day it helps a lot on the productivity side.


```suggestion
const cleanText = () => {
	return inputText.value.toLowerCase().replace(/[^\w\s]/g, ' ').replace(/\s+/g, ' ').trim();
};
const textToWords = () => {
	const text = cleanText();
	return text.split(' ').filter(word => word.length > 0);
};
const generateKeywords = () => {
  if (!inputText.value.trim()) {
    keywords.value = [];
    return;
  }

  const words = textToWords();
  const uniqueWords = [...new Set(words)];
  
  uniqueWords.forEach((word, index) => {
  	oneGramKeywords.value.push(word);
  	
  	if(uniqueWords[index+1]) {
  		const twoGramKeyword = `${word} ${uniqueWords[index+1]}`
  		twoGramKeywords.value.push(twoGramKeyword);
  	}
  	
  	if(uniqueWords[index+2]) {
  		const threeGramKeyword = `${word} ${uniqueWords[index+1] ${uniqueWords[index+2]}}`
  		threeGramKeywords.value.push(threeGramKeyword);
  	}
  })
};
```

turanibrahim commented on src/pages/VKeywordGenerator.vue:
```diff
@@ -0,0 +1,112 @@
+              {{ keyword }}
+            </MaBadge>
+          </div>
+        </div>
```
You need to update this based on new logic

turanibrahim commented on src/App.vue:
```diff
@@ -1,13 +1,25 @@
+
+<style>
+/* Global app-wide styles can stay here if needed */
+</style>
```
Perfect 👍🏼

turanibrahim commented on .eslintrc.cjs:
```diff
@@ -1,12 +1,18 @@
 /* eslint-env node */
 module.exports = {
```
```
require("@rushstack/eslint-patch/modern-module-resolution")`
```

turanibrahim commented on .eslintrc.cjs:
```diff
@@ -1,12 +1,18 @@
-    'eslint:recommended'
+    'airbnb-base',
+    'airbnb-typescript/base'
   ],
```
Most of the config is correct, well done 👍🏼 We just need to add vue-ts library as well

```bash
yarn add --dev @vue/eslint-config-typescript @rushstack/eslint-patch
```


```suggestion
root: true,
  extends: [
    'eslint:recommended',
    '@typescript-eslint/recommended',
		'plugin:vue/vue3-recommended',
		'@vue/eslint-config-typescript'
    'airbnb-base',
    'airbnb-typescript/base'
],
```

We can look together if you have any issue with new config setup

nagmex210 committed:
refactor: implement code review suggestions for improved code quality

  - Replace CSS with Tailwind @apply directives in VFooter and layout
  components
  - Convert activeTab watch to computed property in VHeader for better
  reactivity
  - Move layout-constants.ts to constants directory for better organization
  - Use kebab-case for Vue component names following Vue best practices
  - Refactor keyword generation logic into smaller, focused functions
  - Separate keyword refs into oneGram, twoGram, and threeGram for better
  state management
  - Update ESLint configuration with Vue TypeScript support and modern
  module resolution
  - Remove unnecessary comments and unused event handlers

turanibrahim commented on src/components/organisms/VHeader.vue:
```diff
@@ -0,0 +1,85 @@
+      <ma-tabs
+        v-model:activeKey="activeTab"
+        type="line"
+        tabPosition="top"
```
```suggestion
        tab-position="top"
```

turanibrahim commented on src/components/organisms/VHeader.vue:
```diff
@@ -0,0 +1,85 @@
+    </div>
+    <div class="header-nav">
+      <ma-tabs
+        v-model:activeKey="activeTab"
```
```suggestion
        v-model:active-key="activeTab"
```

turanibrahim commented on src/components/organisms/VHeader.vue:
```diff
@@ -0,0 +1,85 @@
+const router = useRouter();
+const route = useRoute();
+
+
```
```suggestion
```

turanibrahim commented on src/layouts/default.vue:
```diff
@@ -0,0 +1,26 @@
+<template>
+  <div class="layout">
+    <VHeader />
```
```suggestion
    <v-header />
```

turanibrahim commented on src/layouts/default.vue:
```diff
@@ -0,0 +1,26 @@
+    <main class="main-content">
+      <router-view />
+    </main>
+    <VFooter />
```
```suggestion
    <v-footer />
```

turanibrahim commented on src/pages/VKeywordGenerator.vue:
```diff
@@ -0,0 +1,156 @@
+        </ma-button>
+      </div>
+
+      <div v-if="oneGramKeywords.length > 0 || twoGramKeywords.length > 0 || threeGramKeywords.length > 0" class="results-section">
```
I don't like this kind of long logic code inside template it would be more clean if we can move this to computed instead.

```js
const isResultVisible = oneGramKeywords.length > 0 || twoGramKeywords.length > 0 || threeGramKeywords.length > 0;
```

```html
 <div v-if="isResultVisible" class="results-section">
```

turanibrahim reviewed (approved):
Eline saglik

nagmex210 committed:
Update src/components/organisms/VHeader.vue

Co-authored-by: İbrahim Turan <ibrahimturan002@gmail.com>

nagmex210 committed:
refactor(vue): improve code style and template readability

  - Move complex template logic to computed property in VKeywordGenerator
  - Replace long conditional with isResultVisible computed property
  - Add computed import to support new reactive logic

nagmex210 committed:
feat(ONB-202, ONB-203): enhance keyword generator with UX improvements

  - Add multi-selection checkboxes for n-grams (1-10)
  - Implement automatic stop word removal using existing filterArr
  - Add real-time keyword generation on input/selection changes
  - Replace hard-coded n-gram functions with dynamic generateNGrams
  - Use ActionKit components (ma-input textarea, ma-checkbox-group)
  - Style checkbox selection with grid layout and background
  - Add browserslist configuration for last 2 years browsers
  - Maintain TailwindCSS utility classes throughout

nagmex210 commented:
Abi direk ONB-204 den mi devam edeyim 202 203 vardı bunda
